packages/core/src/client/decorators/smartAccountClient.ts

```diff
@@ -54,19 +60,24 @@ export type BaseSmartAccountClientActions<
     args: SendUserOperationParameters<TAccount>
   ) => Promise<UserOperationStruct>;
   buildUserOperationFromTx: (
-    args: SendTransactionParameters<TChain, TAccount>
+    args: SendTransactionParameters<TChain, TAccount>,
+    overrides?: UserOperationOverrides
   ) => Promise<UserOperationStruct>;
   buildUserOperationFromTxs: (
     args: SendTransactionsParameters<TAccount>
   ) => Promise<BuildUserOperationFromTransactionsResult>;
   checkGasSponsorshipEligibility: (
     args: SendUserOperationParameters<TAccount>
   ) => Promise<boolean>;
+  signUserOperation: (
+    args: SignUserOperationParameters<TAccount>
+  ) => Promise<UserOperationRequest>;
   dropAndReplaceUserOperation: (
     args: DropAndReplaceUserOperationParameters<TAccount>
   ) => Promise<SendUserOperationResult>;
   sendTransaction: <TChainOverride extends Chain | undefined = undefined>(
-    args: SendTransactionParameters<TChain, TAccount, TChainOverride>
+    args: SendTransactionParameters<TChain, TAccount, TChainOverride>,
+    overrides?: UserOperationOverrides
```

Comment:
and same here:

and same here::

```suggestion
    args: SendTransactionParameters<TChain, TAccount> & {overrides?: UserOperationOverrides}
```

Reply:
@moldy530 I can't.

viem public client blocked overriding the param types of some of PublicActions protected methods. 

<img width="1082" alt="Screenshot 2024-02-28 at 6 00 05 PM" src="https://github.com/alchemyplatform/aa-sdk/assets/3278577/ea575553-2561-47aa-96b6-0c739b42d25a">

```
export type Client<
  transport extends Transport = Transport,
  chain extends Chain | undefined = Chain | undefined,
  account extends Account | undefined = Account | undefined,
  rpcSchema extends RpcSchema | undefined = undefined,
  extended extends Extended | undefined = Extended | undefined,
> = Client_Base<transport, chain, account, rpcSchema> &
  (extended extends Extended ? extended : unknown) & {
    extend: <
      const client extends Extended &
        Partial<ExtendableProtectedActions<transport, chain, account>>,
    >(
      fn: (
        client: Client<transport, chain, account, rpcSchema, extended>,
      ) => client,
    ) => Client<
      transport,
      chain,
      account,
      rpcSchema,
      Prettify<client> & (extended extends Extended ? extended : unknown)
    >
  }
```

https://github.com/alchemyplatform/aa-sdk/blob/denniswon/sign-send-raw-user-op/packages/core/node_modules/viem/clients/createClient.ts#L59
```
// Actions that are used internally by other Actions (ie. `call` is used by `readContract`).
// They are allowed to be extended, but must conform to their parameter & return type interfaces.
// Example: an extended `call` action must accept `CallParameters` as parameters,
// and conform to the `CallReturnType` return type.
type ExtendableProtectedActions<
  transport extends Transport = Transport,
  chain extends Chain | undefined = Chain | undefined,
  account extends Account | undefined = Account | undefined,
> = Pick<
  PublicActions<transport, chain, account>,
  | 'call'
  | 'createContractEventFilter'
  | 'createEventFilter'
  | 'estimateContractGas'
  | 'estimateGas'
  | 'getBlock'
  | 'getBlockNumber'
  | 'getChainId'
  | 'getContractEvents'
  | 'getEnsText'
  | 'getFilterChanges'
  | 'getGasPrice'
  | 'getLogs'
  | 'getTransaction'
  | 'getTransactionCount'
  | 'getTransactionReceipt'
  | 'prepareTransactionRequest'
  | 'readContract'
  | 'sendRawTransaction'
  | 'simulateContract'
  | 'uninstallFilter'
  | 'watchBlockNumber'
  | 'watchContractEvent'
> &
  Pick<WalletActions<chain, account>, 'sendTransaction' | 'writeContract'>
```

Reply:
ah gotcha, ok yea that checks out